### DISCUSSION TITLE
Is it possible with you app?

### DISCUSSION THREAD

**Discussion by djgje** [Nain57/Smart-AutoClicker#67] [Q&A] [answered]
Hello,
I'm looking for an app that could automatically do some things on the screen, but not in a fixed position...

Here's what I would like, with action in {} and result in ().
1. Identify if an image is displayed on the screen, or in a predefined area to minimize the ressources use
2. If the image is found, in a variable position -> {click on it} -> (a popup is opened)
3. Wait for 0,5s, loading time
4. In the window, search for an other image in an other area of the screen
5. If the image is found -> {click on it} !!Image can appears multiple time in the area, the app should click on every matches, from 0 to 3 or 4
6. Wait for 4s
7. In the same window, search if a third image
8. If found -> {click on it} -> (Refreshing the popup)
9. Redo from 4 to 8
10. Redo from 4 to 8
11. Redo from 4 to 8
12. In the popup that is still displayed, search for a fourth image, last one
13. If found -> {click on it} -> (popup closed)

This macro should be launched every 20 or 30 seconds, starting with searching the first image (1) and so on if found...

I hope that it's clear enough to ask you help.

You app is really interesting, and it's different that other macro builder that doesn't have the image recognition functionality!

Thank you in advance for your help,

Jerome

**Comment by Nain57** [comment, marked as answer]
Hi.

The current algorithm only detects the conditions images at the position they were captured. In the future (and if I find the time), I want to replace my own algorithm by a real computer vision one (via OpenCv) that will allow this kind of things.

Short answer: no, this is not currently possible, unless you capture every possible image positions.

**Reply by GreatestFool** [reply]
Am I right to assume there have been no significant change regarding this? I'm rather interested in a feature like this myself. There's a game that has golden bugs crawling across the screen that would be nice to be able to click, but since they move it's difficult to press them unless I go absolutely full idiot and make a million copies for the bugs at different positions lol.

**Comment by djgje** [comment]
Hello,

thank you for your answer!!

Ok, for me I'm thinking that I could make it, as the positions are 3 or 4... I'll have a try when I'll have time, but yes, I thing that an other algorithm could be a good thing to open your app to more possibilities!

Anyway, I can imagine the work load it is to develop an app like that, and it's already so kind to share it for free!!

I wish you the best and hope that you'll find the time to continue this project!
